bad-code/backend/src/aws.ts

```diff
@@ -8,32 +8,39 @@ const s3 = new S3({
     endpoint: process.env.S3_ENDPOINT
 })
 export const fetchS3Folder = async (key: string, localPath: string): Promise<void> => {
-    const params = {
-        Bucket: process.env.S3_BUCKET ?? "",
-        Prefix: key
-    }
+    try {
+        const params = {
+            Bucket: process.env.S3_BUCKET ?? "",
+            Prefix: key
+        };
 
-    const response = await s3.listObjectsV2(params).promise()
-    // Bounty $25 to make this function run parallelly
-    if (response.Contents) {
-        for (const file of response.Contents) {
-            const fileKey = file.Key
-            if (fileKey) {
-                const params = {
-                    Bucket: process.env.S3_BUCKET ?? "",
-                    Key: fileKey
-                }
-                const data = await s3.getObject(params).promise()
-                if (data.Body) {
-                    const fileData = data.Body
-                    const filePath = `${localPath}/${fileKey.replace(key, "")}`
-                    //@ts-ignore
-                    await writeFile(filePath, fileData)
+        const response = await s3.listObjectsV2(params).promise();
+        if (response.Contents) {
+            // Use Promise.all to run getObject operations in parallel
+            await Promise.all(response.Contents.map(async (file) => {
+                const fileKey = file.Key;
+                if (fileKey) {
+                    const getObjectParams = {
+                        Bucket: process.env.S3_BUCKET ?? "",
+                        Key: fileKey
+                    };
+
+                    const data = await s3.getObject(getObjectParams).promise();
+                    if (data.Body) {
+                        const fileData = data.Body;
+                        const filePath = `${localPath}/${fileKey.replace(key, "")}`;
+                        
+                        await writeFile(filePath, fileData);
```

Comment:
I might be wrong but using async here makes the code run sequentially right? Or am I wrong?

Reply:
Yep! my bad sir

Reply:
@Mahhheshh Keep the await.

Reply:
not in this case, since this code doesn't run sequentially. 
If the first iteration is awaiting, map moves to the second one.